tests/src/MockHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient;
+
+use GuzzleHttp\Handler\MockHandler as GuzzleMockHandler;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+
+class MockHandler extends GuzzleMockHandler
```

Comment:
No need to do this, you can use Middleware::history() to be able to read the responses.
In Drupal is a bit harder to inject a middleware, but it has been done [successfully here](https://github.com/openeuropa/media_avportal/blob/8.x-1.x/tests/modules/media_avportal_mock/media_avportal_mock.services.yml#L3).

Reply:
We will research this. Nice if we can do it w/o this hack. I was on the pressure to assert outgoing requests, NOT the responses. So, if we can read from somewhere the HTTP calls it's better.

Reply:
done